lib/galaxy/dataset_collections/matching.py

```diff
@@ -65,7 +65,9 @@ def structure(self):
         effective_structure = leaf
         for unlinked_structure in self.unlinked_structures:
             effective_structure = effective_structure.multiply(unlinked_structure)
-        linked_structure = self.linked_structure or leaf
+        linked_structure = self.linked_structure
```

Comment:
can we write like this?
`linked_structure = self.linked_structure if self.linked_structure is not None else leaf`

Reply:
Yes we could do that. It is a bit more concise but it requires dereferences linked_structure from ``self`` twice. I feel it looks better to just fetch an object property once inside a function and depending on the implementation it may be faster to just fetch it once. That said it may not be faster if Python can somehow detect it isn't changing in other threads, etc... and I can understand how some sort of view object properties as local variables and so view it is cleaner to duplicate the expression ``self.linked_structure``.

Reply:
@jmchilton ok thanks! I got it :+1: